filesmanager/filesmanager.py

```diff
@@ -48,20 +132,339 @@ def student_view(self, context=None):
+        """Add a directory to a target directory.
+
+        The new directory will:
+        - Be added to the target directory, if found. Otherwise, an error will be returned.
+        - Be added to the root directory if no target directory is specified.
+        - Have a unique incremental ID.
+        - Have a path composed by the target directory path and the directory name.
+        - Have an empty list of children.
 
-        self.count += 1
-        return {"count": self.count}
+        Arguments:
+            name: the name and path of the directory to be added.
+            path: the path of the target directory where the new directory will be added.
+        """
+        directory_name = data.get("name")
+        path = data.get("path")
+        target_directory = self.get_target_directory(path)
+        if not target_directory:
+            return {
+                "status": "error",
+                "message": "Target directory not found",
+            }
+        target_directory.append(
+            {
+                "name": directory_name,
+                "type": "directory",
+                "path": f"{path}/{directory_name}" if path else directory_name,
+                "metadata": {
+                    "id": self.incremental_directory_id,
+                },
+                "children": [],
+            }
+        )
+        self.incremental_directory_id += 1
+        return {
+            "status": "success",
+            "content": target_directory,
+        }
+
+    @XBlock.handler
+    def upload_files(self, request, suffix=''):  # pylint: disable=unused-argument
+        """Handler for file upload to the course assets.
+
+        Arguments:
+            request: the request object containing the files to be uploaded and the
+            target directory path.
+
+        Returns: the content of the target directory.
+        """
+        # Temporary fix for supporting both contentstore assets management versions (master / Palm)
+        try:
+            from cms.djangoapps.contentstore.views.assets import update_course_run_asset  # pylint: disable=import-outside-toplevel
+        except ImportError:
+            from cms.djangoapps.contentstore.asset_storage_handler import update_course_run_asset  # pylint: disable=import-outside-toplevel
+        path = request.params.get("path")
+        target_directory = self.get_target_directory(path)
+        for type, file in request.params.items():
+            if not type.startswith("file"):
+                continue
+            try:
+                content = update_course_run_asset(self.course_id, file.file)
+                target_directory.append(
+                    {
+                        "name": file.filename,
+                        "type": "file",
+                        "path": f"{path}/{file.filename}" if path else file.filename,
+                        "metadata": self.get_asset_json_from_content(content),
+                    }
+                )
```

Comment:
are we going to support any asset type (including .py and .js files)?

I'm not sure how, but I think if not correctly handled .js files can be a vulnerability vector that we don't want to introduce.

Although, those files are supported by the Assets system, it's important to review how will we handle those files

Reply:
The idea is to support all files as the Files & Uploads pages already do since we're displaying what's inside. This is what I think but I'll ask the PM internally to confirm. I'll let you know!

Reply:
I asked the PM yesterday but haven't got any reply yet